### DISCUSSION TITLE
predict the future out of sample data

### DISCUSSION THREAD

**Discussion by Man1978-scd** [ourownstory/neural_prophet#636] [Q&A - get help using NeuralProphet] [answered]
if I set n_lags=10,i want to predict the future out of sample data, for example, i want to predict the 100 in the future, must set the n_forecasts=100?, I am confused, if I set n_forecasts=100, train the model is very slowly

**Comment by bhausleitner** [comment, marked as answer]
Hey @Man1978-scd, 

yes, this is the correct way to do it, unfortunately this also leads to slower training. 

Let me know if this helps!

**Comment by ourownstory** [comment]
@Man1978-scd 
A tip: when using autoregression, try setting n_lags to a value about 2 times that of n_forecasts.
if your model is training slowly, consider if you really need 100 forecasts (less will most often lead to better results).

**Comment by steplp** [comment]
but I need forecast the future datas, how to do?  for example the tps/min， I need use the before datas forecast the future datas, if use n_forecast=1 it is too slow!
